platform/android/java/build.gradle

```diff
@@ -191,94 +93,122 @@ def templateExcludedBuildTask() {
-                tasks += "copy${capitalizedTarget}AARToAppModule"
-                // Copy the generated aar library files to the bin directory.
-                tasks += "copy${capitalizedTarget}AARToBin"
-                // Copy the prebuilt binary templates to the bin directory.
-                tasks += "copy${capitalizedTarget}BinaryToBin"
+                // Copy the Godot android library archive file into the app module libs directory.
+                // Depends on the library build task to ensure the AAR file is generated prior to copying.
+                String copyAARTaskName = "copy${capitalizedTarget}AARToAppModule"
+                if (tasks.findByName(copyAARTaskName) != null) {
+                    buildTasks += tasks.getByName(copyAARTaskName)
+                } else {
+                    buildTasks += tasks.create(name: copyAARTaskName, type: Copy) {
+                        dependsOn ":lib:assembleTemplate${capitalizedTarget}"
+                        from('lib/build/outputs/aar')
+                        include("godot-lib.template_${target}.aar")
+                        into("app/libs/${target}")
+                    }
+                }
+
+                // Copy the Godot android library archive file into the root bin directory.
+                // Depends on the library build task to ensure the AAR file is generated prior to copying.
+                String copyAARToBinTaskName = "copy${capitalizedTarget}AARToBin"
+                if (tasks.findByName(copyAARToBinTaskName) != null) {
+                    buildTasks += tasks.getByName(copyAARToBinTaskName)
+                } else {
+                    buildTasks += tasks.create(name: copyAARToBinTaskName, type: Copy) {
+                        dependsOn ":lib:assembleTemplate${capitalizedTarget}"
+                        from('lib/build/outputs/aar')
+                        include("godot-lib.template_${target}.aar")
+                        into(binDir)
+                    }
+                }
+
+                // Copy the generated binary template into the Godot bin directory.
+                // Depends on the app build task to ensure the binary is generated prior to copying.
+                String copyBinaryTaskName = "copy${capitalizedEdition}${capitalizedTarget}BinaryToBin"
+                if (tasks.findByName(copyBinaryTaskName) != null) {
+                    buildTasks += tasks.getByName(copyBinaryTaskName)
+                } else {
+                    buildTasks += tasks.create(name: copyBinaryTaskName, type: Copy) {
+                        dependsOn ":app:assemble${capitalizedEdition}${capitalizedTarget}"
+                        from("app/build/outputs/apk/${edition}/${target}")
+                        into(binDir)
+                        include("android_${edition}${capitalizedTarget}.apk")
+                        if (isStandard) {
+                            rename("android_${edition}${capitalizedTarget}.apk", "android_${target}.apk")
```

Comment:
Instead or renaming here, you can update the logic in https://github.com/godotengine/godot/blob/d2956c78ccddeae5a8e466a3423df88276e07270/platform/android/java/app/build.gradle#L221 to configure the name of the generated binaries.

Reply:
I'm not sure what `variant.name` is. Without renaming I was getting `android_monoDev.apk` and/or `android_standardDev.apk` copied in the `/bin` directory. So it seems like `variant.name` already includes the edition and build type.

I wanted to avoid using the `standard` prefix to prevent breaking the current scripts and tasks that rely on this, in the hopes that this PR only affects `mono` users. I assume changing the name will break existing projects with an already generated gradle source template, but perhaps it's already part of the general workflow to re-generate this when upgrading Godot.

Reply:
You can find the properties available in `variant` in https://developer.android.com/reference/tools/gradle-api/7.1/com/android/build/api/variant/ComponentIdentity#buildType:kotlin.String.

Based on those properties, I think the following update should do what we need:
```
applicationVariants.all { variant ->
        variant.outputs.all { output ->
              String filenameSuffix = variant.flavorName == "mono" ? variant.name : variant.buildType;            
              output.outputFileName = "android_${filenameSuffix}.apk"
        }
    }
```

Reply:
Done.